source/extensions/access_loggers/http_grpc/grpc_access_log_impl.cc

```diff
@@ -232,7 +232,7 @@ void HttpGrpcAccessLog::emitLog(const Http::HeaderMap& request_headers,
     auto* tls_properties = common_properties->mutable_tls_properties();
     const auto* downstream_ssl_connection = stream_info.downstreamSslConnection();
 
-    tls_properties->set_tls_sni_hostname(stream_info.requestedServerName());
+    tls_properties->set_tls_sni_hostname(std::string(stream_info.requestedServerName()));
```

Comment:
Note the pattern here may cause additional allocation and copy of string, in this particular case, since protobuf generates `set_tls_sni_hostname(std::string&&)` so the extra copy isn't happening, though if you're invoking a method only has `const std::string&` interface (and the interface might out of our control), the extra copy will be needed if you return string_view.

Reply:
Protobuf at the moment requires taking ownership of the input value AFAICT, since internally it is a string. So there is a copy here no matter what, right? The question is whether there is one more extra copy due to lack of a method that takes a `string_view`. Let me look further into the protobuf  thoughts about its lack of support of `absl::string_view`.

Reply:
You are adding one alloc and copy by invoking std::string explicitly. This is not a specific problem for protobuf, but all libraries that take const std::string& / and const char* as parameter.